src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -261,22 +265,33 @@ impl NonOverlapSubLevelPicker {
                 break;
             }
 
+            overlap_len_and_begins.pop();
+
             // check reverse overlap
-            for reverse_index in (0..last_level_index).rev() {
+            for (reverse_index, (old_overlap_len, old_overlap_begin)) in (0..last_level_index)
+                .rev()
+                .zip_eq_fast(overlap_len_and_begins.into_iter().rev())
+            {
                 let target_tables = &l0[reverse_index].table_infos;
-                let overlap_files = overlap_info.check_multiple_overlap(target_tables);
+                let (overlap_files, new_overlap_begin) =
```

Comment:
Is there a better way to avoid clone `SstableInfo` in `check_multiple_overlap` ? It seems that we only use sst-id in this case.

Reply:
How about return start_index and end_index ? And then we do not need clone any object.

Reply:
> How about return start_index and end_index ? And then we do not need clone any object.

fixed